### PR TITLE
build: avoid llgo test panic on packages without test binaries

### PR DESCRIPTION
**Pull request by cpunion**
## Summary
- fix `llgo test` panic when a package has no linkable `.test` package after ModeTest filtering
- always apply ModeTest filtering before multi-package validation
- return early when filtered package list is empty
- add unit tests for empty-filter result and `-o` validation in test mode

## Repro
Before this change:

```bash
./dev/llgo.sh test ./chore/ardump
```

panics with `index out of range [0] with length 0` in `internal/build/build.go` due to `initial[0]` access after filtering.

## Validation
- `go test ./internal/build -count=1`
- `./dev/llgo.sh test ./chore/ardump -count=1`
- `./dev/llgo.sh test ./internal/build -count=1`

### CODE REVIEW

File: internal/build/build_test.go [hunk 1]
```diff
@@ -98,6 +99,40 @@ func TestCmpTest(t *testing.T) {
...
+		}
+		_, err := filterTestPackages(initial, "/tmp/out")
+		if err == nil {
+			t.Fatal("expected error for -o with multiple test packages, got nil")
+		}
+		if !strings.Contains(err.Error(), "cannot use -o flag with multiple packages") {
+			t.Fatalf("unexpected error: %v", err)
+		}
+	})
+}
```
[the remark this answers is not in the document]
Comment (cpunion) [reply internal/build/build_test.go:151]: Addressed in 40743cac: added a happy-path case (e.g. ["foo", "foo.test"] => ["foo.test"]) to lock the normal pass-through behavior.

File: internal/build/build.go [hunk 2]
```diff
@@ -492,6 +488,19 @@ func needLink(pkg *packages.Package, mode Mode) bool {
 	return pkg.Name == "main"
 }
 
+func filterTestPackages(initial []*packages.Package, outFile string) ([]*packages.Package, error) {
+	filtered := make([]*packages.Package, 0, len(initial))
```
[the remark this answers is not in the document]
Comment (cpunion) [reply internal/build/build.go]: Addressed in f63bfa2f: switched filtering to in-place (`filtered := initial[:0]`) to avoid extra allocation.
